### PR TITLE
Setting the right auth db name 

### PR DESCRIPTION
**Pull request by leofigy**
## Description

This small fix is related to https://github.com/mongodb/terraform-provider-mongodbatlas/issues/292
just extracting the **auth** db name in case of update minimal change. 

Tested locally (custom build) before updating the reference in the terraform repo 
```
leofigy@DESKTOP-IQOJR0Q:~/mongo/staging> terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # mongodbatlas_database_user.atlasexternal will be created
  + resource "mongodbatlas_database_user" "atlasexternal" {
      + auth_database_name = "$external"
      + aws_iam_type       = "NONE"
      + id                 = (known after apply)
      + project_id         = "REDACTED"
      + username           = "C=US,ST=California,L=CITY,O=COMPANY,OU=UNIT,CN=sampleuserX"
      + x509_type          = "CUSTOMER"

      + labels {
          + key   = "TestUser"
          + value = "External test"
        }

      + roles {
          + collection_name = (known after apply)
          + database_name   = "admin"
          + role_name       = "readWriteAnyDatabase"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

mongodbatlas_database_user.atlasexternal: Creating...
mongodbatlas_database_user.atlasexternal: Creation complete after 1s [id=YXV0aF9kYXRhYmFzZV9uYW1l:JGV4dGVybmFs-cHJvamVjdF9pZA==:NWNmNWE0NWE5Y2NmNjQwMGU2MDk4MWI2-dXNlcm5hbWU=:Qz1VUyxTVD1DYWxpZm9ybmlhLEw9Q0lUWSxPPUNPTVBBTlksT1U9VU5JVCxDTj1zYW1wbGV1c2VyWA==]


 ~ update in-place

Terraform will perform the following actions:

  # mongodbatlas_database_user.atlasexternal will be updated in-place
  ~ resource "mongodbatlas_database_user" "atlasexternal" {
        auth_database_name = "$external"
        aws_iam_type       = "NONE"
        id                 = "cHJvamVjdF9pZA==:NWNmNWE0NWE5Y2NmNjQwMGU2MDk4MWI2-dXNlcm5hbWU=:Qz1VUyxTVD1DYWxpZm9ybmlhLEw9Q0lUWSxPPUNPTVBBTlksT1U9VU5JVCxDTj1zYW1wbGV1c2VyWA==-YXV0aF9kYXRhYmFzZV9uYW1l:JGV4dGVybmFs"
        project_id         = "5cf5a45a9ccf6400e60981b6"
        username           = "C=US,ST=California,L=CITY,O=COMPANY,OU=UNIT,CN=sampleuserX"
        x509_type          = "CUSTOMER"

        labels {
            key   = "TestUser"
            value = "External test"
        }

      - roles {
          - database_name = "admin" -> null
          - role_name     = "readWriteAnyDatabase" -> null
        }
      + roles {
          + collection_name = (known after apply)
          + database_name   = "admin"
          + role_name       = "atlasAdmin"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Terraform will perform the following actions:

  # mongodbatlas_database_user.atlasexternal will be updated in-place
  ~ resource "mongodbatlas_database_user" "atlasexternal" {
        auth_database_name = "$external"
        aws_iam_type       = "NONE"
        id                 = "dXNlcm5hbWU=:Qz1VUyxTVD1DYWxpZm9ybmlhLEw9Q0lUWSxPPUNPTVBBTlksT1U9VU5JVCxDTj1zYW1wbGV1c2VyWA==-YXV0aF9kYXRhYmFzZV9uYW1l:JGV4dGVybmFs-cHJvamVjdF9pZA==:NWNmNWE0NWE5Y2NmNjQwMGU2MDk4MWI2"
        project_id         = "REDACTED"
        username           = "C=US,ST=California,L=CITY,O=COMPANY,OU=UNIT,CN=sampleuserX"
        x509_type          = "CUSTOMER"

        labels {
            key   = "TestUser"
            value = "External test"
        }

      - roles {
          - database_name = "admin" -> null
          - role_name     = "readWriteAnyDatabase" -> null
        }
      + roles {
          + collection_name = (known after apply)
          + database_name   = "admin"
          + role_name       = "atlasAdmin"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

```

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by themantissa** [comment]
@leofigy did you mean to cite the #292 bug?

**Comment by themantissa** [comment]
Also would suggest specifying this impacts **auth** database name and not database name (I know it's a bit confusing, which is how I think we had an issue w/ this as we added more than SCRAM-SHA auth!)

**Comment by leofigy** [comment]
Hi @themantissa , right. Sorry I updated to point the right issue #292 and fixed the tittle :).

**Review by themantissa — DISMISSED** [review]
LGTM but want to have an internal dev give it a look as well.  Will add and once they approve good on our side. Thank you!!

**Comment by themantissa** [comment]
@gssbzn could you or one of your team give this a quick review/check?

**Review by coderGo93 — DISMISSED** [review]
if the testacc has been passed then LGTM!!

File: mongodbatlas/database_users.go
Comment (gssbzn) [inline mongodbatlas/database_users.go:62]: [nit] does this has any use in the terraform provider? if not I'd rather not export it

File: mongodbatlas/database_users.go
Comment (gssbzn) [inline mongodbatlas/database_users.go:186]: [non-blocking] I'd rather solve this the same way we approached `Delete` where the `databaseName` param was added, we already broke this service before when we did that may as well break it again

**Review by gssbzn — CHANGES_REQUESTED** [review]
One major thing is that LDAP is is also `$external` so right now with this we are falling to account for it. This is if you decide to keep the current `Update` backward compatible but my other comment is that I'd rather pass the database name as a param (same as we do with delete) and let the logic fall into consumers of the method (we do this for the cli) but this is non blocking if we consider keeping this backward compatible

File: mongodbatlas/database_users.go
Comment (leofigy) [reply mongodbatlas/database_users.go:61]: hey thanks

File: mongodbatlas/database_users.go
Comment (leofigy) [reply mongodbatlas/database_users.go:62]: Right it's not used there, but if we break the backward compatibility , because the parameter is optional, the plugin will need to figure it out this, calling this function.

Comment (gssbzn) [reply mongodbatlas/database_users.go:62]: how are you guys doing with delete? flagging in case you may have the same bug for delete as delete you have to pass the authdb
I'm ok with exporting this if we improve on the doc a little
```
// GetAuthDB determines the authentication database based on the type of user.
// LDAP, X509 and AWSIAM should all use $external.
// SCRAM-SHA should use admin
```
kind of thing

Comment (leofigy) [reply mongodbatlas/database_users.go:62]: okay, checked we cached the id for the delete, and we passed it. just one question in the schema in the terraform side, we have 

`                  "database_name": {
                                Type:          schema.TypeString,
                                Optional:      true,
                                ConflictsWith: []string{"auth_database_name"},
                                Deprecated:    "use auth_database_name instead",
                        },
                        "auth_database_name": {
                                Type:          schema.TypeString,
                                Optional:      true,
                                ConflictsWith: []string{"database_name"},
                        },
`

However if we don't wan to break the backward compatibility in the client by now I could just add in the client a non exported json field , instead of modifying the interface signature and then not export the GetAuthDB, what do you think?
`
AuthDatabaseName string `json:"-"` //optional for auth
`

File: mongodbatlas/database_users.go
Comment (gssbzn) [reply mongodbatlas/database_users.go:186]: we talked this off-line we'll keep this like it is

File: mongodbatlas/database_users.go
Comment (leofigy) [reply mongodbatlas/database_users.go:64]: Hi @gssbzn added the documentation line and below is the LDAP check

**Review by gssbzn — APPROVED** [review]
LGTM thanks fro the fixes

**Pull request merged**
